great_expectations/datasource/fluent/spark_file_path_datasource.py

```diff
@@ -1,87 +1,481 @@
+    )
+    # mode: Optional[str] = None,
+    mode: Optional[Literal["PERMISSIVE", "DROPMALFORMED", "FAILFAST"]] = None
+    # columnNameOfCorruptRecord: Optional[str] = None,
+    column_name_of_corrupt_record: Optional[str] = Field(
+        None, alias="columnNameOfCorruptRecord"
+    )
+    # dateFormat: Optional[str] = None,
+    date_format: Optional[str] = Field(None, alias="dateFormat")
+    # timestampFormat: Optional[str] = None,
+    timestamp_format: Optional[str] = Field(None, alias="timestampFormat")
+    # multiLine: Optional[Union[bool, str]] = None,
+    multi_line: Optional[Union[bool, str]] = Field(None, alias="multiLine")
+    # allowUnquotedControlChars: Optional[Union[bool, str]] = None,
+    allow_unquoted_control_chars: Optional[Union[bool, str]] = Field(
+        None, alias="allowUnquotedControlChars"
+    )
+    # lineSep: Optional[str] = None,
+    line_sep: Optional[str] = Field(None, alias="lineSep")
+    # samplingRatio: Optional[Union[float, str]] = None,
+    sampling_ratio: Optional[Union[float, str]] = Field(None, alias="samplingRatio")
+    # dropFieldIfAllNull: Optional[Union[bool, str]] = None,
+    drop_field_if_all_null: Optional[Union[bool, str]] = Field(
+        None, alias="dropFieldIfAllNull"
+    )
+    # encoding: Optional[str] = None,
+    encoding: Optional[str] = None
+    # locale: Optional[str] = None,
+    locale: Optional[str] = None
+    # pathGlobFilter: Optional[Union[bool, str]] = None,
+    # Inherited from _SparkGenericFilePathAssetMixin
+    # recursiveFileLookup: Optional[Union[bool, str]] = None,
+    # Inherited from _SparkGenericFilePathAssetMixin
+    # modifiedBefore: Optional[Union[bool, str]] = None,
+    # Inherited from _SparkGenericFilePathAssetMixin
+    # modifiedAfter: Optional[Union[bool, str]] = None,
+    # Inherited from _SparkGenericFilePathAssetMixin
+    # allowNonNumericNumbers: Optional[Union[bool, str]] = None,
+    allow_non_numeric_numbers: Optional[Union[bool, str]] = Field(
+        None, alias="allowNonNumericNumbers"
+    )
+    # ^^^ spark parameters for pyspark.sql.DataFrameReader.json() (ordered as in pyspark v3.4.0)
+
+    # vvv Docs <> Source Code mismatch
+    # The following parameters are mentioned in https://spark.apache.org/docs/latest/sql-data-sources-json.html
+    # however do not appear in the source code https://github.com/apache/spark/blob/v3.4.0/python/pyspark/sql/readwriter.py#L309
+    # timezone: str = Field(alias="timeZone")
+    # timestamp_ntz_format: str = Field(
+    #     "yyyy-MM-dd'T'HH:mm:ss[.SSS]", alias="timestampNTZFormat"
+    # )
+    # enable_date_time_parsing_fallback: bool = Field(
+    #     alias="enableDateTimeParsingFallback"
+    # )
+    # ^^^ Docs <> Source Code mismatch
+
+    class Config:
+        extra = pydantic.Extra.forbid
+        allow_population_by_field_name = True
+        # pyspark types (from pyspark_types) are not currently validated
+        arbitrary_types_allowed = True
```

Comment:
Can I remove this?

Reply:
@anthonyburdi If the tests pass, then I would say, "Go for it!"